File: src/Projects/P23.MetaTrader4.Manager.ClrWrapper/PerformanceInfo.cpp
#include "stdafx.h"
#include "P23.MetaTrader4.Manager.ClrWrapper.h"

IList<P23::MetaTrader4::Manager::Contracts::PerformanceInfo^>^ P23::MetaTrader4::Manager::ClrWrapper::PerformaneRequest(UInt32 from)
{
	int total = 0;
	PerformanceInfo* infos = _manager->Manager->PerformaneRequest(from, &total);
	IList<P23::MetaTrader4::Manager::Contracts::PerformanceInfo^>^ result = gcnew List<P23::MetaTrader4::Manager::Contracts::PerformanceInfo^>();
	for (int i = 0; i < total; i++)
		result->Add(Convert(&infos[i]));
	return result;
}

File: src/P23.MetaTrader4.Manager.Contracts/PerformanceInfo.cs
using System;

namespace P23.MetaTrader4.Manager.Contracts
{
    /// <summary>
    /// Object that represents performance information
    /// </summary>
    public class PerformanceInfo
    {
        /// <summary>
        /// Time
        /// </summary>
        public UInt32 Ctm { get; set; }

        /// <summary>
        /// Online users
        /// </summary>
        public short Users { get; set; }

        /// <summary>
        /// Processor loading (%)
        /// </summary>
        public short Cpu { get; set; }

        /// <summary>
        /// Free memory (Kb)
        /// </summary>
        public int FreeMem { get; set; }

        /// <summary>
        /// Network activity (Kb/s)
        /// </summary>
        public int Network { get; set; }

        /// <summary>
        /// All open sockets in system
        /// </summary>
        public int Sockets { get; set; }
    };
}
